src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -67,14 +71,61 @@ CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::Write(const ConcreteDataAttribu
     switch (aPath.mAttributeId)
     {
     case Attributes::DefaultOtaProviders::Id: {
+        return WriteDefaultOtaProviders(aPath, aDecoder);
+    }
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::ReadDefaultOtaProviders(AttributeValueEncoder & aEncoder)
+{
+    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    return requestor->GetDefaultOtaProviderList(aEncoder);
+}
+
+CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::WriteDefaultOtaProviders(const ConcreteDataAttributePath & aPath,
+                                                                          AttributeValueDecoder & aDecoder)
+{
+    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    switch (aPath.mListOp)
+    {
+    case ConcreteDataAttributePath::ListOperation::ReplaceAll: {
         DataModel::DecodableList<OtaSoftwareUpdateRequestor::Structs::ProviderLocation::DecodableType> list;
```

Comment:
So we treat 'replace-all' as 'clear'? Is that OK?

Reply:
With chunking, this is how it's done...